substrate/frame/session/src/lib.rs

```diff
@@ -423,7 +423,10 @@ pub mod pallet {
 	#[pallet::genesis_config]
 	#[derive(frame_support::DefaultNoBound)]
 	pub struct GenesisConfig<T: Config> {
+		/// Initial list of validator at genesis representing by their `(AccountId, ValidatorId, Keys)`.
 		pub keys: Vec<(T::AccountId, T::ValidatorId, T::Keys)>,
+		/// List of (AccountId, ValidatorId, Keys) that will be registered at genesis, but not as active validators.
+		pub non_authority_keys: Vec<(T::AccountId, T::ValidatorId, T::Keys)>,
```

Comment:
I disagree, `non_authority_keys` are not necessarily authorities, whether they will be authorities it will be decided by `SessionManager` (even in the session #2). Just for having keys registered they will not be authorities, unless they `SessionManager` decides to select them.

As for the naming, I am fine with anything as I am not too biased around it.

Reply:
Well. 
1. I agree that these are not necessarity authorities, indeed I put in **bold** that are **candidates**.
2. This is not my opinion, is just a fact. The `inner_set_keys` is called, which calls `put_keys`, which stores your `non_authority_keys` in `NextKeys` set, which are the candidates **registered**

Reply:
ah got it. Yeah I can add more clarity on that for sure!

Reply:
Alright so I added a comment saying that these keys are **valid at least until session 2** (maybe valid is not the best word here?), but basically my intention is to say that if there is a session-key change, the first session that would pick these new keys is session number 3